### ISSUE TITLE
Persist groups for timeout

### ISSUE THREAD

**Issue by mattdodge** [nio-blocks/signal_timeout#6]
It would be nice to have a toggle to persist the groups that need timeouts. I think the persistence mixin would make this easy. Right now, we require a signal to come through the block before we start waiting for the timeout

**Comment by shadetree01010100** [comment]
I guess since we can't count on @hansmosh for anything anymore I'll take this one :P
